### ISSUE TITLE
g++-4.9  -pedantic and __int128

### ISSUE THREAD

**Issue by eddelbuettel** [google/cctz#5]
I am considering wrappnig this for use by R. The CRAN maintainers are pretty strict about compiler warnings, and the code in your repo currently leads to two screen fulls of 

```
In file included from ../inst/include/src/cctz_impl.h:23:0,
                 from cctz_cnv.cc:20:
../inst/include/src/cctz_info.h:46:12: warning: ISO C++ does not support ‘__int128’ for ‘offset’ [-Wpedantic]
   __int128 offset;  // seconds from some epoch DateTime
            ^
../inst/include/src/cctz_info.h:57:67: warning: ISO C++ does not support ‘__int128’ for ‘operator-’ [-Wpedantic]
 inline __int128 operator-(const DateTime& lhs, const DateTime& rhs) {
                                                                   ^
../inst/include/src/cctz_info.h:126:59: warning: ISO C++ does not support ‘__int128’ for ‘offset’ [-Wpedantic]
                      int hour, int min, int sec, __int128 offset) const;
                                                           ^
```

I didn't see an obvious way to accommodate this during build-time.  Would you consider addressing it in source?

**Comment by devjgm** [comment]
FYI, we're not ignoring you. We'll take a look at this and reply back here. Thanks.

**Comment by devbww** [comment]
Thanks Dirk.  And thanks for following up, Greg---I started to look into this yesterday, but got sidetracked.  Anyway, the idea is to just eliminate the 128-bit stuff altogether.  It's only there to ensure that we can accurately go to the boundaries of a 64-bit count of seconds in any time zone, but that also means it probably isn't worth it.  I'll get back to you soon.

**Comment by eddelbuettel** [comment]
> FYI, we're not ignoring you.

==:-)  

I am aware that Boost Date_Time has a configuration / compile-time option to use 96 bit representation to allow for nanoseconds (as opposed to 64 bits and micros).  So I was just thinking you're extra precise.  One though I had and meant to try (but haven't) is to drop in `uint128_t` from [this Boost MultiPrecision header](http://www.boost.org/doc/libs/1_59_0/libs/multiprecision/doc/html/boost_multiprecision/tut/ints/cpp_int.html).  You may not like the extra dependency; I was just looking as a test as we happen to have Boost headers for R packages if need be.

BTW I cobbled together a [proof of concept RcppCCTZ package](https://github.com/eddelbuettel/rcppcctz) as well -- mostly just to see if it built even without your fancy new build tool which I don't have hanging around.

**Comment by eddelbuettel** [comment]
`ping`

Any news?

**Comment by devjgm** [comment]
Yes, we're working on a change that should remove our usage of int128 completely. Likely within the next few days. Please stay tuned.

**Comment by SemenMartynov** [comment]
Hi,
have you succeeded at this task?

**Comment by devbww** [comment]
All the 128-bit integers should now be gone.  Please give it a try, and thanks for your patience.

**Comment by eddelbuettel** [comment]
Nice. Will do.

**Comment by devjgm** [comment]
Also, JFYI: I pulled this snapshot and it seems to work fine on OS X with `-ftrapv` enabled. Thanks, @devbww

**Comment by eddelbuettel** [comment]
And my [RcppCCTZ](https://github.com/eddelbuettel/rcppcctz) wrapper package for R is now on [CRAN](https://cran.r-project.org/web/packages/RcppCCTZ/index.html) so that the [results page](https://cran.r-project.org/web/checks/check_results_RcppCCTZ.html) will show build results too (once the cron jobs for that kick in).

**Comment by devbww** [comment]
Hi Dirk,

I see that your Solaris builds failed on the references to tm_zone and tm_gmtoff from cctz_libc.cc.

Does Solaris have corresponding extensions?  If so, we could add some conditions.  But even then the easiest thing to do would be to disable the "libc:<tzname>" support altogether.

```
diff --git a/src/BUILD b/src/BUILD
index fd46494..323eaa4 100644
--- a/src/BUILD
+++ b/src/BUILD
@@ -9,8 +9,6 @@ cc_library(
         "cctz_impl.h",
         "cctz_info.cc",
         "cctz_info.h",
-        "cctz_libc.cc",
-        "cctz_libc.h",
         "cctz_posix.cc",
         "cctz_posix.h",
         "tzfile.h",
diff --git a/src/cctz_if.cc b/src/cctz_if.cc
index 4f0b53e..095f359 100644
--- a/src/cctz_if.cc
+++ b/src/cctz_if.cc
@@ -15,18 +15,10 @@
 
 #include "src/cctz_if.h"
 #include "src/cctz_info.h"
-#include "src/cctz_libc.h"
 
 namespace cctz {
 
 std::unique_ptr<TimeZoneIf> TimeZoneIf::Load(const std::string& name) {
-  // Support "libc:localtime" and "libc:*" to access the legacy
-  // localtime and UTC support respectively from the C library.
-  if (name.compare(0, 5, "libc:") == 0) {
-    return std::unique_ptr<TimeZoneIf>(new TimeZoneLibC(name.substr(5)));
-  }
-
-  // Otherwise use the "zoneinfo" implementation by default.
   std::unique_ptr<TimeZoneInfo> tz(new TimeZoneInfo);
   if (!tz->Load(name)) tz.reset();
   return std::unique_ptr<TimeZoneIf>(tz.release());
```

**Comment by eddelbuettel** [comment]
Wow. Thanks for the follow-up!  I was sort-of resigned to forgetting Solaris but the CRAN maintainer in charge of Solaris is pretty adamant about it.  

Shall we just #ifdef this this?  Either with an 'on' only for Linux and OS X, or an off for Solaris?

I do use `#ifdef __SUNPRO_CC` in Rcpp in places.

**Comment by eddelbuettel** [comment]
And @jeroenooms with Solaris VM once again safes the day.  It would appear that your suggested changes -- as implemented in [this commit](https://github.com/eddelbuettel/rcppcctz/commit/3bc0a5fd052082db6b38dec658d1e2172455d582) -- did the trick.

**Comment by eddelbuettel** [comment]
And [it worked on Solaris too](https://www.r-project.org/nosvn/R.check/r-patched-solaris-sparc/RcppCCTZ-00check.html) -- so thanks again to @devbww for the suggested changed and @jeroenooms for initial testing.

**Comment by devbww** [comment]
Concerning the problem with cctz_libc.cc on Solaris:

We believe we have addressed the issue, but don't have a Solaris platform on which to test.  If you have any reason to want to grab the latest version and give it a try, we'd be very interested in hearing the results.  Thanks.

**Comment by eddelbuettel** [comment]
@devbww: Thanks for the heads-up. If you want to contact me regarding [RcppCCTZ](https://github.com/eddelbuettel/rcppcctz) do feel free to open issue tickets there too.

I incorporated your changes. I am not quite sure where we were with Solaris. I seem to recall we had very early ones which you all addressed. So when I updated [RcppCCTZ](https://github.com/eddelbuettel/rcppcctz)  last weekend, its status actuall _was_ ok: No reported issues.

In any event, it looks clean like a whistle now too: [build reports](https://cloud.r-project.org/web/checks/check_results_RcppCCTZ.html) for all R platforms (that supported C++11, some will be added in the spring when the Windows toolchain gets a long-awaited updated).  

So thanks for all the help -- everything looks pretty good.
